Add Storyscript 0.25 updates (#144)
storyscript/docs.storyscript.io: storyscript/advanced/README.md

```coffeescript
try
  a = dangerousService calculate
finally
  b = 1

c = b # OK
```

### Null value

Every variable in Storyscript MAY be without a value and thus have the empty value `null`.

```coffeescript
revenue = asterioidMining()
if revenue != null

>   reporting send revenue: revenue

Reviewer: ```suggestion
  reporting send revenue:revenue
```

```

The empty value `null` has the following properties:

- `null` can be assigned to all variables
- all variables can be equivalence compared against `null`

However, `null` comes with many runtime restrictions:

- all operations with `null` will result in a runtime error
- Array or Map index access when either the object or key is `null` will result in a runtime error
- calling built-ins or functions on a `null` object will result in a runtime error

> - using `null` as an argument to a service/function/built-in will result in a runtime error if the argument type is not `any`

Reviewer 2: from Slack:
<jude>: Shouldn’t be right? I could pass null to a function expecting a string type
<jude>: since this string type can be null
<jude>: and the function can handle it

Author: Resolved via internal discussion. For future reference, see e.g. that the runtime is already doing this check for services

https://github.com/storyscript/runtime/blob/dafb805cdc81436671735b63677821f6419bdac2/storyruntime/processing/Services.py#L398-L403

## Type conversions

Storyscript provides the `to` operator for type conversions which allows to convert a variable from one type to another type.

For example:

```coffeescript
"1" to int # 1
[1,2] to string # "[1,2]"
[1,2] to List[string] # ["1","2"]
```

In particular, `to` allows convenient handling of services which return `any`:

```
any_var = AnyTypedService call
b = any_var to int + 2
```

However, conversions may fail at runtime:

```coffeescript
"a" to int # ERROR
```

> Like function calls and built-ins in Storyscript, `to` will result in a copy of the object and further operations on the converted object will not modify the original object.

Reviewer: ```suggestion
Like function calls and builtins in Storyscript, `to` will result in a copy of the object and further operations on the converted object will not modify the original object.
```

We should use one or the other I guess

Author: Quick grep:

![image](https://user-images.githubusercontent.com/4370550/67008441-c1bdd300-f0e9-11e9-91f3-9d86e90e892d.png)

-> will go with `built-in` as this seems to be the current majority. Updated the other "builtins".

{"a":"b:"} < {"c": "d"} # Always disallowed
```

### Equality operations

All real types can be checked for their equality with themselves.
Only the virtual `none` type (e.g. from a function without a return type) can't
check for its equality.

> Additionally, is allowed to compare all types with `null`:

Reviewer: ```suggestion
Additionally, it is allowed to compare all types with `null`:
```
